lib/std/math/Interval.zig

```diff
@@ -0,0 +1,110 @@
+const std = @import("../std.zig");
+
+/// Create an interval of type T.
+/// See https://en.wikipedia.org/wiki/Interval_(mathematics) for more info.
+pub fn Interval(comptime T: type) type {
+    comptime std.debug.assert(std.meta.trait.isNumber(T)); // Interval must be instantiated with a number type.
+    return struct {
+        /// The left hand side of the Interval.
+        /// n as null here represents infinity.
+        /// open determines if the interval should be open (exclusive) or closed (inclusive). 
+        from: Endpoint,
+        /// The right hand side of the Interval.
+        /// n as null here represent infinity.
+        /// open determines if the interval should be open (exclusive) or closed (inclusive). 
+        to: Endpoint,
+
+        pub const Endpoint = union(enum) {
+            Infinity,
+            Closed: T,
+            Open: T,
+
+            pub fn n(self: Endpoint) ?T {
```

Comment:
why `n`? (leave a doc-comment?)

Reply:
good catch, this isn't super self explanatory unless you look at the code. it's a helper to get the `T` value out of `.Open` or `.Closed` if those are set or null if its `.Infinity`